Piskvorky

MonikaMudr opened the pull request:
Rozdelila jsem finkce piskvorek do modulu, Vytvorila jsem funkci tah_pocitace, ktera ma v sobe strategii. Napsala jsem testy (test_piskvorky). V test_piskvorky mam problem, ze mi nejde naimportovat pytest. Proto vsehny negativni testy, ktere jsem napsala jsou ve ''', protoze nefunguji. Zkousela jsem si dohledat, kde by mohl byt problem, ale zatim bez uspechu. Nenapada te, cim by to mohlo byt? Soubor piskvorky.py je puvodni hra bez strategie. Upraveny soubor je piskvorky_strategie.

MonikaMudr committed:
Rozdeleni hry piskvorky do modulu

MonikaMudr committed:
Funkce tah pridana do modulu ai, aby nedochazelo k cyklickemu
importu. Vytvoreny testy na jednotlive funkce z piskvorek

ondrejbenes reviewed (commented):
Testy i strategie jsou super, jen jsem trochu zmatený z "organizace projektu" :slightly_smiling_face: 

- Má se hra spouštět přes `hra.py`? Není to jen zapomenutý soubor? Pokud ne, tak proč se tam importuje tolik věcí? Nestačilo by jen naimportvat funkci `piskvorky1D`?
- Necheš soubor `piskvorky.py` smazat, nebo alespoň zakomentovat? Neboj, v historii gitu kdyžtak zůstane. Příjde mi, že byl nahrazen souborem `piskvorky_strategie`
- Funkci `tah` definuješ v `ai.py` i `piskvorky_strategie.py`. Byl bych pro ji jen naimportovat z `ai.py`
  - Zvážila jsi variantu dát funkci `tah` do nového modulu, třeba `utils`? Sice hrozí, že by se z toho pak stalo trochu smetiště, ale vzhledem k tomu, že ta funkce neřeší umělou inteligenci, mi do modulu `ai` úplně nesedí :wink: Možná nedělat modul `utils`, ale třeba `pole`, ve kterém by byly funkce, které mění nebo vyhodnocují herní pole?

Ohledně testů - můžeš poslat do komentáře, jakou ti ty vyhazuje chybu, když zkoušíš `pytest.raises`? Tipl bych, že buď není pytest nainstalovaný ve tvém virtuálním prostředí, nebo test nespouštíš s aktivovaným prostředím :slightly_smiling_face:

MonikaMudr commented:
Co se tyka importu, tak v tom mam trochu zmatek. Hrozne dlouho mi to nefungovalo, protoze jsem tam mela ten cyklicky import, takze jsem zkousela co se da. Pokud to bude fungovat jen jako import piskvorky1D(), tak je to urcite super :-).
Co se tyka te chyby importu, tak primo v Visual Code mi to pise "Unable to import 'pytest'pylint(import-error)". Pokud spustim ten test, tak testy co jsou predtim projdou a na tom, ktery pouziva pytest.raises se to sekne. Musim dat ctrl + c, aby se to ukoncilo a chybu to nehlasi. Pokud spustim jen tu samotnou funkci, tak pise:
NameError: name 'pytest' is not defined.
A spoustim to prikazem "python -m pytest -v test_piskvorky.py", virtualni prostredi mam zaple. A pytest je nainstalovany ve slozce venv\Scripts 
Jeste jsem se chtela zeptat ohledne gitu. Chapu to tak, ze pokud udelam "merge pull request", tak se mi spoji vetev piskvorky s Master? Pokud to chci ale jeste pred merge upravit, je mozne upravit v mem lokalnim repozitari, udelat commit a git push origin piskvorky, a pridaji se tyto zmeny do tohoto pull request? Ci je potreba udelat novy pull request?

ondrejbenes commented:
- `import piskvorky1D` by mělo jít bez problému :wink: 
- testy s `pytest.raises` se ti "seknou", protože cyklus ve funci `tah_pocitace_bez_strategie` běží nekonečně dlouho - zkus v té funkci udělat, aby se vyhodila výjimka, když pole neobsahuje "-"
- chápeš to správně, github je chytrý, když uděláš push do větve, ze které je pull request, tak se do toho pull requesty ty změny přidají

MonikaMudr committed:
Smazani souboru piskvorky(soubor bez strategie), smazany zbytecne importy
v modulu hra.py

MonikaMudr committed:
Upraveny jednotlive moduly. Pridan modul pole, ktery obsahuje vsechny funkce,
ktere upravuji pole. Upraveny importy

MonikaMudr committed:
Upravena funkce tah_pocitace_bez_strategie tak, aby se v pripade chybejiciho
volneho pole vyvolala vyjimka.
Zkontrolovana funkcnost vsech testu z modulu test_piskvorky.py
